### PR TITLE
PP-4097 migrate all copy to json

### PR DESCRIPTION
**Pull request by jonheslop**
So we can show pages translated into other languages we need all the strings served from a JSON file which we can switch out.

Most of the strings in the happy payments path were already in but none of the error ones were

### CODE REVIEW

File: app/views/auth_3ds_required.njk [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {% extends "layout.njk" %}
 
-{% block pageTitle %}Payment in progress{% endblock %}
+{% block pageTitle %}{{ i18n.errorViews.inProgress }}{% endblock %}
```
Comment (alexbishop1) [inline app/views/auth_3ds_required.njk]: This isn’t an error: 3D Secure being required is a normal part of the flow

File: app/views/auth_waiting.njk [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% extends "layout.njk" %}
 
 {% block pageTitle %}
-  Processing your payment
+  {{ i18n.errorViews.processingPayment }}
```
Comment (alexbishop1) [inline app/views/auth_waiting.njk]: Again, not really an error

File: locales/en.json [hunk 3]
```diff
@@ -80,17 +81,73 @@
...
+    "viewSummary": "View your payment summary",
+    "paymentExpired": "Your Payment expired",
+    "sessionExpired": "Your payment session has expired",
+    "paymentCancelled": "Your Payment was cancelled",
+    "paymentCancelledTitle": "Payment cancelled",
+    "paymentCancelledDescription": "Your payment has been cancelled by the government service. This may have happened for a number of reasons.",
+    "error": "An error occurred",
+    "technicalProblems": "Sorry, we’re experiencing technical problems"
+  },
+  "otherViews": {
```
Comment (alexbishop1) [inline locales/en.json]: With the exception of `inProgress`, `userCancelledPayment` and `goBackToService`, I think these are all basically shown during the authorisation part of the process (including 3D Secure), so maybe we could move most of them into a `i18n.authorisation` object and try to find homes for the others (I’m worried that `otherViews` will just become a grab bag where people put anything and everything).

(Yes, I realise that some of the strings are shown in the `capture_waiting.njk` page but I think that page dates back to before we had asynchronous capture and could probably be removed at some point.)

File: app/views/charge.njk [hunk 4]
```diff
@@ -71,25 +71,31 @@
           {% endif %}
         </ul>
         <p class="govuk-body-s accepted-cards-hint withdrawal-text">
-          {{ withdrawalText }}
+          {% if withdrawalText === 'debit_credit' %}
+            {{ i18n.cardDetails.withdrawalText.debit_credit }}
+          {% endif %}
+          {% if withdrawalText === 'debit' %}
+            {{ i18n.cardDetails.withdrawalText.debit }}
+          {% endif %}
```
[the remark this answers is not in the document]
Comment (jonheslop) [reply app/views/charge.njk]: Before hand we were building the object path within the charge controller which was difficult to understand:
https://github.com/alphagov/pay-frontend/blob/master/app/controllers/charge_controller.js#L41-L42

Also it buries a string in a controller when all the others come from the template so it makes sense to have it here.

File: app/views/auth_3ds_required_out.njk [hunk 5]
```diff
@@ -5,14 +5,14 @@
...
-  <p class="govuk-body-l lede">To continue with your payment, we need your bank’s approval.</p>
+  <p class="govuk-body-l lede">{{ i18n.authorisation.approvalNeeded }}</p>
 </noscript>
 <form name="three_ds_required" method="post" action="{{ issuerUrl }}">
   <input type="hidden" name="PaReq" value="{{ paRequest }}"/>
   <input type="hidden" name="MD" value="{{ md }}"/>
   <input type="hidden" name="TermUrl" value="{{ threeDSReturnUrl }}"/>
   <noscript>
-    <button id="confirm" class="govuk-button">Continue</button>
+    <button id="confirm" class="govuk-button">{{ i18n.commonButtons.continueButton}}</button>
```
Comment (alexbishop1) [inline app/views/auth_3ds_required_out.njk]: Space before `}}`?

File: app/views/error_with_return_url.njk [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
 {% block content %}
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
     <h3 class="govuk-heading-l">
-      An error occurred:
+      {{ i18n.errorViews.error }}:
     </h3>
     <p class="govuk-body-l lede">
-      <a id="return-url" class="govuk-link" href="{{ returnUrl }}">Return to service</a>
+      <a id="return-url" class="govuk-link" href="{{ returnUrl }}">{{ i18n.authorisation.goBackToService }}</a>
```
Comment (alexbishop1) [inline app/views/error_with_return_url.njk]: I’d missed that this was used here: maybe move this to `i18n.commonLinks.goBackToService`?

File: app/views/errors/incorrect_state/auth_3ds_required.njk [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
   <div class="govuk-grid-column-two-thirds">
-    <h1 class="govuk-heading-l">Your payment is in progress</h1>
-    <p class="govuk-body-l lede">You can either:</p>
-    <a class="govuk-body-l lede" href="/card_details/{{ chargeId }}/3ds_required" id="threeds-required-link">Continue with your payment</a>
-    <p class="govuk-body-l lede">or<br/>
-      <a href="{{ returnUrl }}" id="return-url" class="govuk-link">Go back to try the payment again</a>
+    <h1 class="govuk-heading-l">{{ i18n.authorisation.inProgress }}</h1>
+    <p class="govuk-body-l lede">{{ i18n.errorViews.either }}</p>
+    <a class="govuk-body-l lede" href="/card_details/{{ chargeId }}/3ds_required" id="threeds-required-link">{{ i18n.errorViews.continue }}</a>
+    <p class="govuk-body-l lede">{{ i18n.cardDetails.or }}<br/>
```
Comment (alexbishop1) [inline app/views/errors/incorrect_state/auth_3ds_required.njk]: Another one I missed: `i18n.commonConjunctions.or`? Or is that too ridiculous?

File: app/views/errors/incorrect_state/auth_success.njk [hunk 8]
```diff
@@ -1,18 +1,18 @@
 {% extends "../../layout.njk" %}
 
 {% block pageTitle %}
-Your payment is in progress - GOV.UK Pay
+  {{ i18n.authorisation.inProgress }} - GOV.UK Pay
```
Comment (alexbishop1) [inline app/views/errors/incorrect_state/auth_success.njk]: I think I suggested `i18n.commonHeadings.inProgress` to keep this out of `i18n.authorisation`

**Pull request merged**